cpp/src/io/statistics/column_stats.cu

```diff
@@ -276,7 +276,9 @@ gatherFloatColumnStats(stats_state_s *s, statistics_dtype dtype, uint32_t t, Sto
       warp_reduce(storage.float_stats[t / 32]).Reduce(s->warp_max[t & 0x1f].fp_val, cub::Max());
     if (!(t & 0x1f)) { s->ck.max_value.fp_val = (vmax != 0.0) ? vmax : CUDART_ZERO; }
   } else if (t < 32 * 3) {
-    vsum = warp_reduce(storage.float_stats[t / 32]).Sum(s->warp_sum[t & 0x1f].fp_val);
+    double val = s->warp_sum[t & 0x1f].fp_val;
+    if (isnan(val)) { val = 0; }
+    vsum = warp_reduce(storage.float_stats[t / 32]).Sum(val);
```

Comment:
This makes it consistent with the change above and removes the additional state.
```suggestion
    auto const& val = s->warp_sum[t & 0x1f].fp_val;
    vsum = warp_reduce(storage.float_stats[t / 32]).Sum(isnan(val) ? 0 : val);
```